src/Lykke.AlgoStore.Api/Infrastructure/ContentFilters/RequiredPermissionFilter.cs

```diff
@@ -25,14 +25,15 @@ public void OnActionExecuting(ActionExecutingContext context)
             var requiredPermission = (context.ActionDescriptor as ControllerActionDescriptor)?.ActionName;
 
             // Check if the action needs a permission
-            var needsPermission = context.Controller.GetType().GetMethods().Where(m => m.ReturnType == typeof(Task<IActionResult>) && m.Name == requiredPermission).FirstOrDefault().GetCustomAttributes(typeof(RequirePermission), false) != null;
+            var reflectedMethod = context.Controller.GetType().GetMethods().Where(m => m.ReturnType == typeof(Task<IActionResult>) && m.Name == requiredPermission).FirstOrDefault();
+            var needsPermission = reflectedMethod.GetCustomAttributes(typeof(RequirePermissionAttribute), false).FirstOrDefault() != null || reflectedMethod.ReflectedType.GetCustomAttributes(typeof(RequirePermissionAttribute), false).FirstOrDefault() != null;
 
-            if(needsPermission)
+            if (needsPermission)
             {
                 var hasPermission = false;
                 for (var i = 0; i < userRoles.Count; i++)
```

Comment:
You can completely remove  for loop and use LINQ
var hasPermission = userRoles.Any(x => x.Permissions.Any(y => y.Name == requiredPermission));